libcontainer/libcontainer_fuzzer.go

```diff
@@ -0,0 +1,85 @@
+// +build gofuzz
+
+package libcontainer
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/opencontainers/runc/libcontainer/configs"
+)
+
+func checkRootInData(data []byte, dir string) error {
+	for i := 0; i < len(dir); i++ {
+		if !strings.Contains(string(data), dir[:i]) {
+			return errors.New(dir[:i] + " not found in " + string(data))
+		}
+	}
+	return nil
+}
+
+func FuzzStateApi(data []byte) int {
+	if len(data) < 3 {
+		return -1
+	}
+	if strings.Contains(string(data), "null") {
+		return -1
+	}
+	root, err := newTestRoot()
+	if err != nil {
+		return -1
+	}
+
+	// The path to our root dir MUST be found
+	// in our input
+	err = checkRootInData(data, root)
+	if err != nil {
+		return -1
+	}
+	var config *configs.Config
```

Comment:
Why pointer?

Reply:
in the helper `newContainerWithName` in the fuzzer we call into f.Create (https://github.com/opencontainers/runc/blob/master/libcontainer/factory_linux.go#L243) which expects a pointer.